src/IConnection.h

```diff
@@ -62,7 +62,7 @@ namespace sdbus::internal {
                                                   , void* userData ) = 0;
 
         virtual PlainMessage createPlainMessage() const = 0;
-        virtual MethodCall createMethodCall( const std::string& destination
+        virtual MethodCall createMethodCall( const std::optional<std::string>& destination
```

Comment:
I'm not sure whether we can change the public API in this way. I see two options:
1. Add a separate `createMethodCall` method which does not have the first parameter
2. Keep the API as it is, but accept an empty `destination` for p2p connections.

Given how similar libraries (QtDBus and GIO) have implemented this, I think that the second option might be preferable.

Reply:
This derived class is in `sdbus::internal` namespace and i think it is not public API. Because it's internal, i guess it's not bad to change arguments in methods that doesn't affect clients code.
In `include/sdbus-c++/IProxy.h`  i've added 3 overloads of the `createProxy` method, because it's public api and i don't want to change methods that clients have already used.

Reply:
Ah, silly me, you are right! OK, then, I'm still thinking that `std::optional` is a bit overkill, but let's leave it up for the maintainer to decide.